packages/progress-bar/src/ProgressBar/ProgressBar.tsx

```diff
@@ -36,14 +39,41 @@ export function ProgressBar(props: ProgressBarProps) {
 
   const { theme } = useDarkMode(darkMode);
 
-  const showIcon = iconsPendingCompletion.includes(color)
+  const showIcon = ICONS_PENDING_COMPLETION.includes(color)
     ? showIconProp && value === maxValue
     : showIconProp;
 
+  // set HTML attributes based on type
   const role = type === 'meter' ? 'meter' : 'progressbar';
-
   const progressBarId = `${role}-${getNodeTextContent(label) || 'default'}`;
 
+  // if progress bar was previously indeterminate and is turning determinate, apply fade-out transition
+  const [animationMode, setAnimationMode] = useState<AnimationMode>(
+    getAnimationMode({
+      type,
+      isIndeterminate,
+      enableAnimation,
+    }),
+  );
+
+  useEffect(() => {
+    if (animationMode === AnimationMode.Indeterminate && !isIndeterminate) {
```

Comment:
Something else I missed is the `usePrevious` hook, which you can use to store previous prop values. [Check out this example from `Code`](https://github.com/mongodb/leafygreen-ui/blob/main/packages/code/src/LanguageSwitcher/LanguageSwitcher.tsx#L20).  With this, you can store the value of `isIndeterminate` and check if the previous value is different from the new value. This way, you're checking the same prop.

Also, it's probably unlikely that someone will do this, but will this check cause any unexpected behavior if the `type` prop changes?